rust/ip-packet/src/proptests.rs

```diff
@@ -0,0 +1,81 @@
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+use pnet_packet::Packet;
+use proptest::arbitrary::any;
+use proptest::strategy::Strategy;
+
+use crate::make::{tcp_packet, udp_packet};
+use crate::MutableIpPacket;
+
+fn non_icmp_packet() -> impl Strategy<Value = MutableIpPacket<'static>> {
+    (
+        any::<IpAddr>(),
+        any::<IpAddr>(),
+        any::<u16>(),
+        any::<u16>(),
+        any::<Vec<u8>>(),
+        any::<bool>(),
+    )
+        .prop_filter("IPs must be of the same version", |(src,dst,..)| src.is_ipv4() == dst.is_ipv4())
+        .prop_map(|(src, dst, sport, dport, payload, is_tcp)|
+            // using a bool instead of prop_oneof because we can't wrap mutableippacket in Just because it's non-clonable
+            if is_tcp {
+                 tcp_packet(src, dst, sport, dport, payload)
+             } else {
+                 udp_packet(src, dst, sport, dport, payload)
+             }
+         )
+}
+
+#[test_strategy::proptest()]
+fn can_translate_dst_packet(
```

Comment:
Can we also add a test (or replace these?) with one that just converts back and forth and checks that the packets are identical?

Technically, those tests could be cheated by just not implementing anything in the translation but we know that we don't do that and I think it would make the tests simpler because we are asserting that the two functions are the inverse of each other. That also has to be true for ICMP packets so we can include those in the tests too.